diff --git a/miniaudio.h b/miniaudio.h
index cbfab74..db425d5 100644
--- a/miniaudio.h
+++ b/miniaudio.h
@@ -20471 +20471 @@ static void ma_device__handle_data_callback(ma_device* pDevice, void* pFramesOut
-                    ma_clip_samples_f32((float*)pFramesOut, (const float*)pFramesOut, frameCount * pDevice->playback.channels);   /* Intentionally specifying the same pointer for both input and output for in-place processing. */
+                    ma_clip_samples_f32((float*)pFramesOut, (const float*)pFramesOut, (ma_uint64)frameCount * pDevice->playback.channels);   /* Intentionally specifying the same pointer for both input and output for in-place processing. */
@@ -29488 +29488 @@ static ma_result ma_device_init_by_type__alsa(ma_device* pDevice, const ma_devic
-        ma_snd_pcm_uframes_t actualBufferSizeInFrames = ma_calculate_buffer_size_in_frames_from_descriptor(pDescriptor, internalSampleRate, pConfig->performanceProfile) * internalPeriods;
+        ma_snd_pcm_uframes_t actualBufferSizeInFrames = (ma_snd_pcm_uframes_t)ma_calculate_buffer_size_in_frames_from_descriptor(pDescriptor, internalSampleRate, pConfig->performanceProfile) * internalPeriods;
@@ -29540 +29540 @@ static ma_result ma_device_init_by_type__alsa(ma_device* pDevice, const ma_devic
-        bufferBoundary = internalPeriodSizeInFrames * internalPeriods;
+        bufferBoundary = (ma_snd_pcm_uframes_t)internalPeriodSizeInFrames * internalPeriods;
@@ -33088 +33088 @@ static int ma_device__jack_buffer_size_callback(ma_jack_nframes_t frameCount, vo
-        size_t newBufferSize = frameCount * (pDevice->capture.internalChannels * ma_get_bytes_per_sample(pDevice->capture.internalFormat));
+        size_t newBufferSize = (size_t)frameCount * ((size_t)pDevice->capture.internalChannels * ma_get_bytes_per_sample(pDevice->capture.internalFormat));
@@ -33101 +33101 @@ static int ma_device__jack_buffer_size_callback(ma_jack_nframes_t frameCount, vo
-        size_t newBufferSize = frameCount * (pDevice->playback.internalChannels * ma_get_bytes_per_sample(pDevice->playback.internalFormat));
+        size_t newBufferSize = (size_t)frameCount * ((size_t)pDevice->playback.internalChannels * ma_get_bytes_per_sample(pDevice->playback.internalFormat));
@@ -33263 +33263 @@ static ma_result ma_device_init__jack(ma_device* pDevice, const ma_device_config
-        pDevice->jack.pIntermediaryBufferCapture = (float*)ma_calloc(pDescriptorCapture->periodSizeInFrames * ma_get_bytes_per_frame(pDescriptorCapture->format, pDescriptorCapture->channels), &pDevice->pContext->allocationCallbacks);
+        pDevice->jack.pIntermediaryBufferCapture = (float*)ma_calloc((size_t)pDescriptorCapture->periodSizeInFrames * ma_get_bytes_per_frame(pDescriptorCapture->format, pDescriptorCapture->channels), &pDevice->pContext->allocationCallbacks);
@@ -33315 +33315 @@ static ma_result ma_device_init__jack(ma_device* pDevice, const ma_device_config
-        pDevice->jack.pIntermediaryBufferPlayback = (float*)ma_calloc(pDescriptorPlayback->periodSizeInFrames * ma_get_bytes_per_frame(pDescriptorPlayback->format, pDescriptorPlayback->channels), &pDevice->pContext->allocationCallbacks);
+        pDevice->jack.pIntermediaryBufferPlayback = (float*)ma_calloc((size_t)pDescriptorPlayback->periodSizeInFrames * ma_get_bytes_per_frame(pDescriptorPlayback->format, pDescriptorPlayback->channels), &pDevice->pContext->allocationCallbacks);
@@ -43987 +43987 @@ MA_API ma_result ma_device_init(ma_context* pContext, const ma_device_config* pC
-            intermediaryBufferSizeInBytes = pDevice->playback.intermediaryBufferCap * ma_get_bytes_per_frame(pDevice->playback.format, pDevice->playback.channels);
+            intermediaryBufferSizeInBytes = (ma_uint64)pDevice->playback.intermediaryBufferCap * ma_get_bytes_per_frame(pDevice->playback.format, pDevice->playback.channels);
@@ -58798 +58798 @@ MA_API ma_result ma_pcm_rb_acquire_read(ma_pcm_rb* pRB, ma_uint32* pSizeInFrames
-    sizeInBytes = *pSizeInFrames * ma_pcm_rb_get_bpf(pRB);
+    sizeInBytes = (size_t)*pSizeInFrames * ma_pcm_rb_get_bpf(pRB);
@@ -58815 +58815 @@ MA_API ma_result ma_pcm_rb_commit_read(ma_pcm_rb* pRB, ma_uint32 sizeInFrames)
-    return ma_rb_commit_read(&pRB->rb, sizeInFrames * ma_pcm_rb_get_bpf(pRB));
+    return ma_rb_commit_read(&pRB->rb, (size_t)sizeInFrames * ma_pcm_rb_get_bpf(pRB));
@@ -58827 +58827 @@ MA_API ma_result ma_pcm_rb_acquire_write(ma_pcm_rb* pRB, ma_uint32* pSizeInFrame
-    sizeInBytes = *pSizeInFrames * ma_pcm_rb_get_bpf(pRB);
+    sizeInBytes = (size_t)*pSizeInFrames * ma_pcm_rb_get_bpf(pRB);
@@ -58844 +58844 @@ MA_API ma_result ma_pcm_rb_commit_write(ma_pcm_rb* pRB, ma_uint32 sizeInFrames)
-    return ma_rb_commit_write(&pRB->rb, sizeInFrames * ma_pcm_rb_get_bpf(pRB));
+    return ma_rb_commit_write(&pRB->rb, (size_t)sizeInFrames * ma_pcm_rb_get_bpf(pRB));
@@ -69129,2 +69129,2 @@ static MA_INLINE ma_uint64 ma_noise_read_pcm_frames__white(ma_noise* pNoise, voi
-        const ma_uint32 bps = ma_get_bytes_per_sample(pNoise->config.format);
-        const ma_uint32 bpf = bps * channels;
+        const ma_uint64 bps = ma_get_bytes_per_sample(pNoise->config.format);
+        const ma_uint64 bpf = bps * channels;
@@ -69248,2 +69248,2 @@ static MA_INLINE ma_uint64 ma_noise_read_pcm_frames__pink(ma_noise* pNoise, void
-        const ma_uint32 bps = ma_get_bytes_per_sample(pNoise->config.format);
-        const ma_uint32 bpf = bps * channels;
+        const ma_uint64 bps = ma_get_bytes_per_sample(pNoise->config.format);
+        const ma_uint64 bpf = bps * channels;
@@ -69330,2 +69330,2 @@ static MA_INLINE ma_uint64 ma_noise_read_pcm_frames__brownian(ma_noise* pNoise,
-        const ma_uint32 bps = ma_get_bytes_per_sample(pNoise->config.format);
-        const ma_uint32 bpf = bps * channels;
+        const ma_uint64 bps = ma_get_bytes_per_sample(pNoise->config.format);
+        const ma_uint64 bpf = bps * channels;
@@ -73762 +73762 @@ MA_API ma_result ma_node_graph_init(const ma_node_graph_config* pConfig, const m
-        pNodeGraph->pProcessingCache = (float*)ma_malloc(pConfig->processingSizeInFrames * pConfig->channels * sizeof(float), pAllocationCallbacks);
+        pNodeGraph->pProcessingCache = (float*)ma_malloc((size_t)pConfig->processingSizeInFrames * pConfig->channels * sizeof(float), pAllocationCallbacks);
@@ -73865,2 +73865,2 @@ MA_API ma_result ma_node_graph_read_pcm_frames(ma_node_graph* pNodeGraph, void*
-            MA_COPY_MEMORY(pRunningFramesOut, pNodeGraph->pProcessingCache, framesToReadFromCache * channels * sizeof(float));
-            MA_MOVE_MEMORY(pNodeGraph->pProcessingCache, pNodeGraph->pProcessingCache + (framesToReadFromCache * channels), (pNodeGraph->processingCacheFramesRemaining - framesToReadFromCache) * channels * sizeof(float));
+            MA_COPY_MEMORY(pRunningFramesOut, pNodeGraph->pProcessingCache, (size_t)framesToReadFromCache * channels * sizeof(float));
+            MA_MOVE_MEMORY(pNodeGraph->pProcessingCache, pNodeGraph->pProcessingCache + ((size_t)framesToReadFromCache * channels), (size_t)(pNodeGraph->processingCacheFramesRemaining - framesToReadFromCache) * channels * sizeof(float));
@@ -74369 +74369 @@ static ma_result ma_node_input_bus_read_pcm_frames(ma_node* pInputNode, ma_node_
-                    float* pPreMixBuffer = (float*)ma_stack_alloc(((ma_node_base*)pInputNode)->pNodeGraph->pPreMixStack, preMixBufferCapInFrames * inputChannels * sizeof(float));
+                    float* pPreMixBuffer = (float*)ma_stack_alloc(((ma_node_base*)pInputNode)->pNodeGraph->pPreMixStack, (size_t)preMixBufferCapInFrames * inputChannels * sizeof(float));
@@ -74647 +74647 @@ static ma_result ma_node_get_heap_layout(ma_node_graph* pNodeGraph, const ma_nod
-            cachedDataSizeInBytes += cacheCapInFrames * ma_get_bytes_per_frame(ma_format_f32, pConfig->pInputChannels[iBus]);
+            cachedDataSizeInBytes += (size_t)cacheCapInFrames * ma_get_bytes_per_frame(ma_format_f32, pConfig->pInputChannels[iBus]);
@@ -74651 +74651 @@ static ma_result ma_node_get_heap_layout(ma_node_graph* pNodeGraph, const ma_nod
-            cachedDataSizeInBytes += cacheCapInFrames * ma_get_bytes_per_frame(ma_format_f32, pConfig->pOutputChannels[iBus]);
+            cachedDataSizeInBytes += (size_t)cacheCapInFrames * ma_get_bytes_per_frame(ma_format_f32, pConfig->pOutputChannels[iBus]);
@@ -75531 +75531 @@ static ma_result ma_node_read_pcm_frames(ma_node* pNode, ma_uint32 outputBusInde
-    ma_apply_volume_factor_f32(pFramesOut, totalFramesRead * ma_node_get_output_channels(pNodeBase, outputBusIndex), ma_node_output_bus_get_volume(&pNodeBase->pOutputBuses[outputBusIndex]));
+    ma_apply_volume_factor_f32(pFramesOut, (ma_uint64)totalFramesRead * ma_node_get_output_channels(pNodeBase, outputBusIndex), ma_node_output_bus_get_volume(&pNodeBase->pOutputBuses[outputBusIndex]));
@@ -76949 +76949 @@ static void ma_engine_node_process_pcm_frames__general(ma_engine_node* pEngineNo
-                    ma_copy_pcm_frames(pRunningFramesOut, pWorkingBuffer, framesJustProcessedOut * channelsOut, ma_format_f32, channelsOut);
+                    ma_copy_pcm_frames(pRunningFramesOut, pWorkingBuffer, (ma_uint64)framesJustProcessedOut * channelsOut, ma_format_f32, channelsOut);
@@ -76952 +76952 @@ static void ma_engine_node_process_pcm_frames__general(ma_engine_node* pEngineNo
-                    ma_copy_and_apply_volume_factor_f32(pRunningFramesOut, pWorkingBuffer, framesJustProcessedOut * channelsOut, volume);
+                    ma_copy_and_apply_volume_factor_f32(pRunningFramesOut, pWorkingBuffer, (ma_uint64)framesJustProcessedOut * channelsOut, volume);
@@ -76960 +76960 @@ static void ma_engine_node_process_pcm_frames__general(ma_engine_node* pEngineNo
-                    ma_apply_volume_factor_f32(pRunningFramesOut, framesJustProcessedOut * channelsOut, volume);
+                    ma_apply_volume_factor_f32(pRunningFramesOut, (ma_uint64)framesJustProcessedOut * channelsOut, volume);
@@ -78584 +78584 @@ static ma_result ma_sound_init_from_data_source_internal(ma_engine* pEngine, con
-        pSound->pProcessingCache = (float*)ma_calloc(pSound->processingCacheCap * ma_get_bytes_per_frame(ma_format_f32, engineNodeConfig.channelsIn), &pEngine->allocationCallbacks);
+        pSound->pProcessingCache = (float*)ma_calloc((size_t)pSound->processingCacheCap * ma_get_bytes_per_frame(ma_format_f32, engineNodeConfig.channelsIn), &pEngine->allocationCallbacks);
@@ -83029 +83029 @@ MA_API ma_uint64 ma_dr_wav_read_pcm_frames_le(ma_dr_wav* pWav, ma_uint64 framesT
-        bytesToRead = (MA_SIZE_MAX / bytesPerFrame) * bytesPerFrame;
+        bytesToRead = (ma_uint64)(MA_SIZE_MAX / bytesPerFrame) * bytesPerFrame;
